test-data/example.jsx

```diff
@@ -30,7 +30,7 @@ const pgettext = (ctx, msg) =>
 const npgettext = (ctx, msg, msgpl, n) =>
   TRANSLATE ? jedTranslated.npgettext(ctx, msg, msgpl, n) : jedEmpty.npgettext(ctx, msg, msgpl, n);
 
-const {Translate, PluralTranslate} = makeComponents({gettext, ngettext, pgettext, npgettext});
+const {Translate, PluralTranslate} = makeComponents(gettext, ngettext, pgettext, npgettext);
```

Comment:
how does this not break things? the argument we pass there (also when [using it in indico](https://github.com/indico/indico/blob/f0765e717cf16283a8ae3ba2705a8699276c6aad/indico/web/client/js/react/i18n.js#L11)) is the `jed` instance, which has those functions as methods.

Reply:
So the method signature is actually `export const makeComponents = (...args) => {`
and there's this whole logic in `getGettextFuncs` to handle different arguments. You can either pass a single object with the functions as properties or you can pass the functions individually (though this is only used in the tests)

Reply:
huh, i wonder what i was thinking when doing that back then :D

Reply:
Well we _could_ simplify the signature and only allow a single object. That would just require a couple small changes to the tests where we call `makeComponents`

Reply:
hmm maybe better if we only want to support jed anyway (and then in tests we simply pass something that looks enough like jed for this library to work)

Reply:
I updated it to only take a single argument (the jed instance)